roles/swift-common/handlers/main.yml

```diff
@@ -1,12 +1,29 @@
 ---
-- name: pip install swifttool
-  pip: name="/opt/stack/{{ swift.swifttool_dir  }}" extra_args='-i https://pypi.python.org/simple/'
+- name: pip install swifttool - source
+  pip: name="/opt/stack/{{ swift.swifttool_dir  }}" extra_args="-i https://pypi.python.org/simple/"
+       virtualenv="{{ openstack_source.virtualenv_base }}/swift"
   register: result
   until: result|success
   retries: 3
   delay: 10
   notify:
     - update ca-certs
+    - pip clean swifttool outside venv
+
+- name: pip install swifttool - package
+  pip: name="/opt/stack/{{ swift.swifttool_dir  }}" extra_args="-i https://pypi.python.org/simple/"
+       virtualenv="{{ openstack_package.virtualenv_base }}/swift"
+  register: result
+  until: result|success
+  retries: 3
+  delay: 10
+  notify:
+    - update ca-certs
+    - pip clean swifttool outside venv
```

Comment:
So I think you can have a single pip install task, that for the `virtualenv` value just use

`/opt/openstack/current/swift/`  That will work for both source and package as the paths are now the same.